core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1050,7 +1050,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
    * }}}
    */
   final def resurrect(implicit ev1: E <:< Throwable): RIO[R, A] =
-    self.sandbox.mapError(_.squash)
+    self.unrefineWith({ case e => e })(ev1)
```

Comment:
Fine to do it the other way here if you think that is more efficient.

Reply:
Using `foldCauseM` is like taping directly into `new ZIO.Fold` so might be hard to beat that!
However, reusing existing combinators improves the discovery of those!